src/features/game/expansion/Land.tsx

```diff
@@ -199,8 +207,16 @@ const getIslandElements = ({
                 name={name}
                 id={building.id}
                 index={itemIndex}
-                readyAt={building.readyAt}
-                createdAt={building.createdAt}
+                readyAt={
+                  name in BUILDING_UPGRADES && game[buildingKey].upgradeReadyAt
+                    ? game[buildingKey].upgradeReadyAt
+                    : building.readyAt
+                }
+                createdAt={
+                  name in BUILDING_UPGRADES && game[buildingKey].upgradedAt
+                    ? game[buildingKey].upgradedAt
+                    : building.createdAt
+                }
```

Comment:
It might be cleaner to assign these values to consts above. Ternary's are much more readable when used on cone 

```ts
// Example
const hasUpgraded = name in BUILDING_UPGRADES && game[buildingKey].upgradeReadyAt
const readyAt = hasUpgraded ? game[buildingKey].upgradeReadyAt : building.readyAt
```